src/modules/musig/session_impl.h

```diff
@@ -678,8 +678,12 @@ int secp256k1_musig_partial_sign(const secp256k1_context* ctx, secp256k1_musig_p
     /* Fails if the magic doesn't match */
     ret = secp256k1_musig_secnonce_load(ctx, k, &pk, secnonce);
     /* Set nonce to zero to avoid nonce reuse. This will cause subsequent calls
-     * of this function to fail */
-    memset(secnonce, 0, sizeof(*secnonce));
+     * of this function to fail. Declassify as memclear marks the bytes
+     * as undefined. */
+    secp256k1_memclear(secnonce, sizeof(*secnonce));
+    secp256k1_declassify(ctx, secnonce->data, sizeof(secp256k1_musig_secnonce_magic));
+    secp256k1_declassify(ctx, &secnonce->data[68], 64);
```

Comment:
We do not need to declassify because we do not use `secnonce` anymore after this.

Reply:
Ok I understand now. I missed that `secp256k1_memclear` uses the UNDEFINED macro (in VERIFY builds).

I think it's indeed preferrable to explicitly "redefine" the macro in the test instead of declassifying in partial_sign. I think the purpose of UNDEFINEing is to catch accidental use of the buffer and there's no legitimate reason to read the secnonce buffer except in the test that checks that it was cleared correctlya1.